syntax = "proto3";

service FerService {
    rpc FER (ImagesArray) returns (EmotionsArray);
}

message ImageRequest {
    bytes image_data = 1;
}

message Emotion {
    string label = 1;
    float confidence = 2;
}

message EmotionsArray {
    repeated Emotion emotions = 1;
}

message ImagesArray {
    repeated ImageRequest images = 1;
}